GLOSSARY.md

```diff
@@ -0,0 +1,128 @@
+
+```
+f(false, false) f(false, true) f(true, false) f(true, true).
+```
+
+As a shorthand the resulting truth table is encoded with `true = 1` and `false =
+0`.
+
+```
+| Truth table | Operator symbol | Common name                      |
+|-------------|-----------------|----------------------------------|
+| 0001        | &&              | Conjunction; and                 |
+| 0111        | ||              | Disjunction; or                  |
+| 1001        | <=>             | Equivalence; if and only if; iff |
+| 1101        | =>              | Implication; if ... then         |
+```
+
+## Binder
+A binder is an expression that binds a variable to a certain expression.
+Examples for binders:
+
+- The universal quantifier `forall(a)` states that a certain condition holds for
+  all allowed values for `a`.
+- A function definition `f(x) = a * x` is a binder for the variable `x` whereas
+  `a` is a free variable.
+- A sum `\sum_n x_n` binds the index variable `n`.
+
+## Canonical
+## Clause
+In the A clause is the disjunction of several expressions. For example the clause
+`condition_1 || condition_2 || ...` states that at least one of the conditions
+holds.
+
+There are two notable special cases of clauses. A *Horn clause* has at most one
+positive literal. A *Definite clause* has exactly one positive literal.
+
+## Datum
+(empty)
+
+## DeBrujin Index
+(empty)
+
+## Formula
+A formula is a logical expression consisting of literals and constants connected
+by logical operators.
+
+## Goal
+With a set of given types, traits and impls, a goal specifies a problem where
+types need to be found that satisfy the problem.
+
+## Literal
+A literal is an atomic element of a formula together with the constants `true`
+and `false`. It is equivalent to a variable in an algebraic expressions.
+
+## Normal form
+To say that a statement is in a certain *normal form* means that the pattern in
+which the subformulas are arranged fulfil certain rules. The individual patterns
+have different advantages for their manipulation.
+
+### Conjunctive normal form (CNF)
+A formula in CNF is a disjunction of conjunctions. For example `(x1 || x2 ||
+x3) && (x4 || x5 || x6)` is in CNF.
+
+### Disjunctive normal form (DNF)
+A formula in DNF is a conjunction of disjunctions. For example `(x1 && x2 &&
+x3) || (x4 && x5 && x6)` is in DNF.
+
+### Negation normal form (NNF)
+A formula in NNF consists only of literals, the connectives `&&` and `||` and
+`true` and `false`.
+
+### Prenex normal form (PNF)
+All quantifiers are on the highest level of a formula and do not occur inside
+the subformulas of the expression.
+
+- `forall(x). exists(y). forall(z). P(x) && P(y) => P(z)` is in PNF.
+- `(exists(x). P(x)) => exists(y). P(y) && forall(z). P(z)` is *not* in PNF.
+
+## Normalization
+Normalization is the process of converting an associated type to a concrete
+type.
+
+## Projection
+Projection is the reference to a field or (in the context of Rust) to a type.
+
```

Comment:
I'm a bit unsure about *Projection*... I understand what *normalization* is and added an example accordingly. Can you provide an example for this one?